libpromises/evalfunction.c

```diff
@@ -1331,79 +1349,159 @@ static FnCallResult FnCallPackagesMatching(ARG_UNUSED EvalContext *ctx, ARG_UNUS
         GetSoftwarePatchesFilename(filename);
     }
 
-    Log(LOG_LEVEL_DEBUG, "%s: reading inventory from '%s'", fp->name, filename);
-
+    Log(LOG_LEVEL_DEBUG, "Reading inventory from '%s'", filename);
+    
     FILE *const fin = fopen(filename, "r");
     if (fin == NULL)
     {
         Log(LOG_LEVEL_VERBOSE,
-            "%s cannot open the %s packages inventory '%s' - "
+            "Cannot open the %s packages inventory '%s' - "
             "This is not necessarily an error. "
             "Either the inventory policy has not been included, "
-            "or it has not had time to have an effect yet. "
+            "or it has not had time to have an effect yet or you are using"
+            "new package promise and check for legacy promise is made."
             "A future call may still succeed. (fopen: %s)",
-            fp->name,
             installed_mode ? "installed" : "available",
             filename,
             GetErrorStr());
 
-        pcre_free(matcher);
-        return FnFailure();
+        return true;
     }
-
-    JsonElement *json = JsonArrayCreate(50);
-    int linenumber = 0;
+    
     char *line;
-
     while (NULL != (line = GetCsvLineNext(fin)))
     {
-        if (strlen(line) > CF_BUFSIZE - 80)
+        if (!AddPackagesmatchingJsonLine(matcher, json, line))
         {
-            Log(LOG_LEVEL_ERR,
-                "Line %d from package inventory '%s' is too long (%zd) to be sensible",
-                linenumber, filename, strlen(line));
             free(line);
-            break; /* or continue ? */
+            break;
         }
+        free(line);
+    }
 
-        if (StringMatchFullWithPrecompiledRegex(matcher, line))
+    bool ret = feof(fin);
```

Comment:
For consistency I would say `bool ret = (feof(fin) != 0);` here. `feof()` does not return a true bool. This also makes it C++ compatible, should we ever go down that route.